### ISSUE TITLE
Automatically test python to works internally

### ISSUE THREAD

**Issue by latot** [cbm755/octsympy#771]
Hi, is related to this https://github.com/cbm755/octsympy/issues/770, actually to start using python we just try python, but in some systems like in the issue sympy is only available in 2 or 3, so the idea is can detect which version works.

An extension of this would be improve python in sympref but different, when we check the available version of python list them there to can select it, there can be some ideas, list all python versions we can found, or just list python versions with sympy.

Thx. Cya.

**Comment by cbm755** [comment]
Sounds helpful to certain users but maybe a lot of out-of-scope effort for us (especially to make it work Windows/Mac).  Also, I personally dislike OS-level features ending up inside every little tool.  I mean, surely it should be Someone Else's Problem to have a working Python and Sympy install.  Does our little tool really need extra knobs and whistles to workaround the (apparently) hard problem of software installation?

I suppose I might accept a simple patch for the first idea (automatically trying "python" and "python3" to find a sympy installation).

We should also be careful we don't end up with settings that user then expect but are difficult to make work with pytave.

**Comment by mtmiller** [comment]
Yeah my understanding of this was that it would involve just testing "python" and "python3" to find the first one that has a usable SymPy.

And yes, pytave currently only works with a single instance of Python, whichever was selected when the user configured and built it. It may be able to support 2 and 3 at runtime in the future, but not right now.

**Comment by mtmiller** [comment]
Or maybe test all three of "python", "python2", and "python3", in some order, since some distros have already made python be python3.

**Comment by latot** [comment]
I'm agree with that, i'm thinking only as an option, we can automatically select the right python version, but we should be do some way to the user can select what python works?

I mean, lets suppose sympy and all the complements works fine over python2 and 3, so we can do some option to can select what version we want to use, that i think can works in all platforms, and to see what python is we can show it like:

python (3.7)
python2 (2.4)
python3 (3.4)

What do you think?

**Comment by mtmiller** [comment]
That seems out of scope to me. If the user wants to know what `python3` is, they can run `system('python3 -V')`.

If the user wants to use a particular Python, they can `export PYTHON=python3` or `setenv PYTHON python3`.

If the user wants a suitable Python to be detected and used, then they would make sure `PYTHON` is **not** in the environment and let the automatic method proposed here do its job.

**Comment by latot** [comment]
Okis, so then we can send a warning if the default python version don't support some component and continue.
